agent/src/main/java/reactor/blockhound/BlockHound.java

```diff
@@ -86,13 +86,29 @@ public static Builder builder() {
      * @see BlockHound#builder()
      */
     public static void install(BlockHoundIntegration... integrations) {
+        builder(integrations).install();
+    }
+
+    /**
+     * Entrypoint for installation via the {@code -javaagent=} command-line option.
+     *
+     * @param args Options for the agent.
+     * @param instrumentation Instrumentation API.
+     *
+     * @see java.lang.instrument
+     */
+    public static void premain(String[] args, Instrumentation instrumentation) {
+        builder().withInstrumentation(instrumentation).install();
```

Comment:
The `builder()` method won't call the builder(BlockHoundIntegration... integrations) signature (it will instead invoke the [builder with empty parameter](https://github.com/reactor/BlockHound/blob/master/agent/src/main/java/reactor/blockhound/BlockHound.java#L76), which does not load any BlockHoundIntegration services, that's a problem.

as a first work around, we could change:
```
builder().withInstrumentation(instrumentation).install();
```
by
```

builder(new BlockHoundIntegration[0]).withInstrumentation(instrumentation).install();
```

but maybe it's not so nice to do that, and also we now have two builder static methods, maybe we should try to keep only one builder() static method that just instantiates the Builder, and put all builder logic in the Builder methods, so what do you think about just adding a new `loadIntegration` method in the Builder class, just above the `with` method, like this:

```
        /**
         * Loads integrations with {@link ServiceLoader} and adds provided integrations
         * using {{@link #with(BlockHoundIntegration)}}.
         * If you don't want to load the integrations using service loader, only use
         * {@link #with(BlockHoundIntegration)} method.
         *
         * @param integrations an array of integrations to automatically apply on the builder using
         * {@link #with(BlockHoundIntegration)}
         * @return this
         * @see BlockHound#builder()
         */
        public Builder loadIntegrations(BlockHoundIntegration... integrations) {
            ServiceLoader<BlockHoundIntegration> serviceLoader = ServiceLoader.load(BlockHoundIntegration.class);
            Stream
                    .concat(StreamSupport.stream(serviceLoader.spliterator(), false), Stream.of(integrations))
                    .sorted()
                    .forEach(this::with);
            return this;
        }
```

Then the premain() method would become:

```
    public static void premain(String args, Instrumentation instrumentation) {
        builder()
                .loadIntegrations()
                . withInstrumentation(instrumentation)
                .install();
    }
```

The install method would also become:

```
    public static void install(BlockHoundIntegration... integrations) {
        builder()
                .loadIntegrations(integrations)
                .install();
    }
```